src/treq/api.py

```diff
@@ -14,6 +28,7 @@ def head(url, **kwargs):
     return _client(kwargs).head(url, _stacklevel=4, **kwargs)
 
 
+@_like(HTTPClient.get)
```

Comment:
Interesting approach... I have a branch somewhere where I started factoring out TypedDict for all the arguments but it is so painful I haven't worked up the will to finish it.

Reply:
Oh, huh, will Mypy actually do the right thing with a TypedDict in this context?

Reply:
(I do think I prefer this approach regardless, that one had just never occurred to me.)

Reply:
I expect it will given [Unpack](https://docs.python.org/3/library/typing.html#typing.Unpack) (my experience actually doing this is with Pyright).

I certainly agree that this is less painful!